src/main/java/com/aws/greengrass/shadowmanager/ShadowManager.java

```diff
@@ -239,166 +239,178 @@ protected void install(InstallConfig installConfig) {
+                int maxLocalShadowUpdatesPerThingPerSecond = Coerce.toInt(rateLimitsPojo
+                        .get(CONFIGURATION_MAX_LOCAL_REQUESTS_RATE_PER_THING_TOPIC));
+                Validator.validateLocalShadowRequestsPerThingPerSecond(maxLocalShadowUpdatesPerThingPerSecond);
+                inboundRateLimiter.setRate(maxLocalShadowUpdatesPerThingPerSecond);
+            }
+        } catch (InvalidConfigurationException e) {
+            serviceErrored(e);
         }
+    }
 
-        if (installConfig.configureSyncDirectionConfig) {
-            config.lookup(CONFIGURATION_CONFIG_KEY, CONFIGURATION_SYNCHRONIZATION_TOPIC,
-                            CONFIGURATION_SYNC_DIRECTION_TOPIC)
-                    .dflt(Direction.BETWEEN_DEVICE_AND_CLOUD.getCode())
-                    .subscribe((why, newv) -> {
-                        Direction newSyncDirection;
-                        if (WhatHappened.removed.equals(why)) {
-                            newSyncDirection = Direction.BETWEEN_DEVICE_AND_CLOUD;
-                        } else {
-                            try {
-                                newSyncDirection = Direction.getDirection(Coerce.toString(newv));
-                            } catch (IllegalArgumentException e) {
-                                serviceErrored(e);
-                                return;
-                            }
-                        }
-
-                        logger.atDebug()
-                                .setEventType("config")
-                                .kv("syncDirection", newSyncDirection.toString())
-                                .log();
-                        Direction currentDirection = syncHandler.getSyncDirection();
-                        // if the sync direction is the same, then just return and do nothing.
-                        if (newSyncDirection.equals(currentDirection)) {
-                            return;
-                        }
-
-                        setupSync(newSyncDirection, Optional.of(currentDirection));
-                        syncHandler.setSyncDirection(newSyncDirection);
-                    });
+    private void configureMaxSizeDocLimitConfig(Node newv) {
+        if (newv != null && !newv.childOf(CONFIGURATION_MAX_DOC_SIZE_LIMIT_B_TOPIC)) {
+            return;
         }
+        int newMaxShadowSize = Coerce.toInt(config.lookup(CONFIGURATION_CONFIG_KEY,
+                CONFIGURATION_MAX_DOC_SIZE_LIMIT_B_TOPIC)
+                .dflt(DEFAULT_DOCUMENT_SIZE));
+        try {
+            Validator.validateMaxShadowSize(newMaxShadowSize);
+            Validator.setMaxShadowDocumentSize(newMaxShadowSize);
+            logger.atDebug()
+                    .setEventType("config")
+                    .kv("maxShadowSize", newMaxShadowSize)
+                    .log();
+        } catch (InvalidConfigurationException e) {
+            serviceErrored(new InvalidConfigurationException(e));
```

Comment:
does this need to create a `new InvalidConfigurationException`

Should these configureMethods just throw InvalidConfiguration and you can let the method that calls them move the service to `serviceErrored`?

Reply:
In this PR, I'm only addressing the subscription on top level configuration key vs individual keys. Nothing inside of these methods is actually modified. I'd prefer to refactor the code in the later PRs if needed.

Reply:
Updated.